lib/Doctrine/ORM/AbstractQuery.php

```diff
@@ -957,9 +955,12 @@ public function getHints()
      */
     public function iterate($parameters = null, $hydrationMode = null)
     {
-        @trigger_error(
-            'Method ' . __METHOD__ . '() is deprecated and will be removed in Doctrine ORM 3.0. Use toIterable() instead.',
-            E_USER_DEPRECATED
+        Deprecation::trigger(
+            'doctrine/orm',
```

Comment:
It's always going to be `doctrine/orm` here, right? It would be cool to auto-detect that, I did it some time ago in the PHPUnit bridge: https://github.com/symfony/phpunit-bridge/blob/0a67187d0c23cc00a14b6af752d5f8b2412c4f16/DeprecationErrorHandler/Deprecation.php#L213

Reply:
This code looks expensive to execute at runtime/production, I am unsure how to do this "performantly" without resorting to a simple thing such as regexping for `vendor/{package}/` in triggering file. That wouldn't work when run on the root package for example.

Reply:
Ok let's ignore my comment then. I think one of the main selling points of your implementation is to be performant in production, let's try to keep it that way 👍